src/main/java/org/springframework/data/couchbase/SimpleCouchbaseClientFactory.java

```diff
@@ -74,7 +74,7 @@ private SimpleCouchbaseClientFactory(final Supplier<Cluster> cluster, final Stri
 
 	@Override
 	public CouchbaseClientFactory withScope(final String scopeName) {
-		return new SimpleCouchbaseClientFactory(cluster, bucket.name(), scopeName);
+		return new SimpleCouchbaseClientFactory(cluster, bucket.name(), scopeName != null ? scopeName : getScope().name());
```

Comment:
If scopeName is null, 
 the old behavior would return a clientFactory with scope = _default
 the new behavior returns a clientFactory with the same scope as the "this". (yes, it could actually return "this" instead of a new object - that is a better implementation).
 This is to allow just using withScope(scopeName) without checking if scopeName is null.